src/components/DaynamicQrScanner.tsx

```diff
@@ -0,0 +1,96 @@
+/* eslint-disable no-console */
+import React, { useEffect, useMemo, useState } from 'react'
+import { BarCodeScanningResult, Camera } from 'expo-camera'
+import { Linking, Platform, StyleSheet } from 'react-native'
+import { BarCodeScanner } from 'expo-barcode-scanner'
+import { useNavigation } from '@react-navigation/native'
+import { useAsync } from 'react-async-hook'
+import { useTranslation } from 'react-i18next'
+import useAlert from '@hooks/useAlert'
+import { CameraScannerLayout } from './CameraScannerLayout'
+import Box from './Box'
+import BackScreen from './BackScreen'
+import ProgressBar from './ProgressBar'
+import Text from './Text'
+
+type Props = {
+  progress: number
+  onBarCodeScanned: (data: string) => void
+}
+const DaynamicQrScanner = ({ onBarCodeScanned, progress }: Props) => {
+  const [hasPermission, setHasPermission] = useState<boolean>()
+  const navigation = useNavigation()
+  const { showOKCancelAlert } = useAlert()
+  const { t } = useTranslation()
+
+  useEffect(() => {
+    Camera.requestCameraPermissionsAsync().then(({ status }) => {
+      setHasPermission(status === 'granted')
+    })
+  }, [])
+
+  useAsync(async () => {
+    if (hasPermission !== false) return
+
+    const decision = await showOKCancelAlert({
+      title: t('qrScanner.deniedAlert.title'),
+      message: t('qrScanner.deniedAlert.message'),
+      ok: t('qrScanner.deniedAlert.ok'),
+    })
+
+    if (decision) {
+      if (Platform.OS === 'ios') {
+        Linking.openURL('app-settings:')
+      } else {
+        Linking.openSettings()
+      }
+    }
+
+    navigation.goBack()
+  }, [hasPermission, navigation, showOKCancelAlert])
+
+  const barCodeScannerSettings = useMemo(
+    () => ({
+      barCodeTypes: [BarCodeScanner.Constants.BarCodeType.qr],
+    }),
+    [],
+  )
+  const handleBarCodeScanned = (result: BarCodeScanningResult) => {
+    onBarCodeScanned(result.data)
+  }
+
+  if (!hasPermission) {
+    return (
+      <BackScreen>
+        <Box />
```

Comment:
Can we display some error to the user vs a blank screen?

Reply:
resolved